Cannot build - Time library is out of date

Currently it's unable to build because of the time library being out of date, more info [here](https://github.com/time-rs/time/issues/696).

```
error[E0282]: type annotations needed for `Box<_>`
  --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.22/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling console v0.15.7
   Compiling time v0.1.45
   Compiling atty v0.2.14
   Compiling csv-core v0.1.10
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

It will successfully build again after running `cargo update time@0.3.22`

If `cargo update` is run in it's entirety then two new errors are thrown:

```
error[E0107]: struct takes 0 lifetime arguments but 1 lifetime argument was supplied
   --> src/ntds/data_table_record.rs:258:69
    |
258 | impl<'info, 'db> From<&DataTableRecord<'info, 'db>> for term_table::Table<'_> {
    |                                                                     ^^^^^---- help: remove these generics
    |                                                                     |
    |                                                                     expected 0 lifetime arguments
    |
note: struct defined here, with 0 lifetime parameters
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/term-table-1.4.0/src/lib.rs:398:12
    |
398 | pub struct Table {
    |            ^^^^^
```
    
and

```
    error[E0308]: mismatched types
   --> src/ntds/data_table.rs:189:65
    |
189 |                         let mut table = term_table::Table::from(&record);
    |                                         ----------------------- ^^^^^^^ expected `Table`, found `&DataTableRecord<'_, '_>`
    |                                         |
    |                                         arguments to this function are incorrect
    |
note: associated function defined here
   --> /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/convert/mod.rs:585:8
   ```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,6 @@ strum = { version = "0.25", features = ["derive", "phf"] }
 num-traits = "0.2.15"
 num-derive = "0.4.0"
 thiserror = "1"
-paste = "1.0"
 bodyfile = "0.1.4"
 hashbrown = "0"
 libesedb = "0.2.5"

--- a/src/esedb_mitigation.rs
+++ b/src/esedb_mitigation.rs
@@ -1,6 +1,6 @@
 use std::io;
 
-#[deprecated(note="this function is a mitigation for some strange behaviour. see https://github.com/janstarke/ntdsextract2/pull/16")]
+//#[deprecated(note="this function is a mitigation for some strange behaviour. see https://github.com/janstarke/ntdsextract2/pull/16")]
 pub(crate) fn libesedb_count(count_fn: impl Fn() -> io::Result<i32>) -> io::Result<i32> {
     match count_fn() {
         Ok(val) => Ok(val),

--- a/src/ntds/data_table_record.rs
+++ b/src/ntds/data_table_record.rs
@@ -255,7 +255,7 @@ impl<'info, 'db> WithValue<ColumnIndex> for DataTableRecord<'info, 'db> {
     }
 }
 
-impl<'info, 'db> From<&DataTableRecord<'info, 'db>> for term_table::Table<'_> {
+impl<'info, 'db> From<&DataTableRecord<'info, 'db>> for term_table::Table {
     fn from(value: &DataTableRecord<'info, 'db>) -> Self {
         let mut table = term_table::Table::new();
         let all_attributes = value.all_attributes();
@@ -263,8 +263,8 @@ impl<'info, 'db> From<&DataTableRecord<'info, 'db>> for term_table::Table<'_> {
         keys.sort();
 
         table.add_row(Row::new(vec![
-            TableCell::new_with_alignment("Attribute", 1, Alignment::Center),
-            TableCell::new_with_alignment("Value", 1, Alignment::Center),
+            TableCell::builder("Attribute").alignment(Alignment::Center).build(),
+            TableCell::builder("Value").alignment(Alignment::Center).build(),
         ]));
 
         for id in keys {

--- a/src/win32_types/timestamp/database_time.rs
+++ b/src/win32_types/timestamp/database_time.rs
@@ -1,4 +1,4 @@
-use chrono::{DateTime, Utc, NaiveDate, NaiveTime, NaiveDateTime};
+use chrono::{DateTime, NaiveDate, NaiveDateTime, NaiveTime, Utc};
 
 use crate::impl_timestamp;
 
@@ -23,7 +23,7 @@ impl From<i64> for DatabaseTime {
             Some(val) => val,
             None => panic!("unable to convert '{bytes:?}' into a database time"),
         };
-        let ts = DateTime::<Utc>::from_utc(NaiveDateTime::new(date, time), Utc);
+        let ts = DateTime::<Utc>::from_naive_utc_and_offset(NaiveDateTime::new(date, time), Utc);
         Self(ts)
     }
 }

--- a/src/win32_types/timestamp/mod.rs
+++ b/src/win32_types/timestamp/mod.rs
@@ -13,7 +13,7 @@ pub use unix_timestamp::*;
 pub use timeline_entry::*;
 
 lazy_static! {
-    static ref BASE_DATETIME: DateTime<Utc> = DateTime::<Utc>::from_utc(
+    static ref BASE_DATETIME: DateTime<Utc> = DateTime::<Utc>::from_naive_utc_and_offset(
         NaiveDate::from_ymd_opt(1601, 1, 1)
             .unwrap()
             .and_hms_opt(0, 0, 0)

